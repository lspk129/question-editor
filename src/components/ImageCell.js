import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageLabel = styled.label`
  border: 1px solid #ccc;
  width: 100%;
  height: 100%;
  cursor: pointer;
  display: grid;
  align-items: center;
  justify-items: center;
  margin: auto;
  font-size: 12px;
`;

const ImageInput = styled.input`
  display: none;
`;

export const Image = styled.img`
  width: 50px;
  height: 50px;
  align-self: center;
  justify-self: center;
`;

export const ImageCell = ({ change, id }) => (
  <ImageLabel>
    <ImageInput
      type={'file'}
      onChange={change}
      data-img={id}
    />
      img
  </ImageLabel>
);

ImageCell.propTypes = {
  change: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};
