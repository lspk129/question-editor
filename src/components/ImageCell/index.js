import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageLabel = styled.label`
  width: 100%;
  height: 100%;
  cursor: pointer;
  display: grid;
  align-items: center;
  justify-items: center;
  margin: auto;
  font-size: 12px;
  ${({ icon }) => icon && `
    background: url(${icon});
    background-size: 30px 30px;
    background-repeat: no-repeat;
    background-position: center;
    cursor: pointer;
    opacity: 0.7;
  `};
`;

const ImageInput = styled.input`
  display: none;
`;

export const Image = styled.img`
  width: 50px;
  height: 50px;
  align-self: center;
  justify-self: center;
  ${({ small }) => small && `
    width: 30px;
    padding: 0 10px 0 20px;
    opacity: 0.7;
  `}
`;

export const ImageCell = ({ change, id, icon }) => (
  <ImageLabel icon={icon} >
    <ImageInput
      type={'file'}
      onChange={change}
      data-img={id}
    />
  </ImageLabel>
);

ImageCell.propTypes = {
  change: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};
