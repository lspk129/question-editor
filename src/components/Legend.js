import React from 'react';
import styled from 'styled-components';
import { Image } from './ImageCell';
import deleteIcon from '../icons/ic_delete_forever_black_24px.svg';
import imageIcon from '../icons/ic_image_black_24px.svg';
import addIcon from '../icons/ic_add_circle_black_24px.svg';

const Section = styled.div`
  grid-column: 2;
  padding: 0 20px 20px;
`;

const ImageRow = styled.p`
  display: flex;
  align-items: center;
  margin: 0;
  ${({ note }) => note && `
    padding: 20px 0 0 30px;
  `};
`;

const Text = styled.span`
  padding-left: 10px;
`;

const Legend = () => (
  <Section>
    <h3>Legend</h3>
    <ImageRow><Image small src={imageIcon} /> - <Text>Add images</Text></ImageRow>
    <ImageRow><Image small src={addIcon} /> - <Text>Add row/column</Text></ImageRow>
    <ImageRow><Image small src={deleteIcon} /> - <Text>Delete row/column</Text></ImageRow>
    <ImageRow note><i>Italic</i> <Text>text is editable</Text></ImageRow>
  </Section>
);

export default Legend;
