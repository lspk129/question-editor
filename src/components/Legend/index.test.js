import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Legend from './';
import { Image } from '../ImageCell';
import deleteIcon from '../../icons/ic_delete_forever_black_24px.svg';
import imageIcon from '../../icons/ic_image_black_24px.svg';
import addIcon from '../../icons/ic_add_circle_black_24px.svg';

describe('Legend component', () => {
  const wrapper = mount(<Legend />,
  );

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has title with h3 tag', () => {
    expect(wrapper.find('h3')).toBePresent();
    expect(wrapper.find('h3')).toHaveText('Legend');
  });

  it('has image component with imageIcon', () => {
    expect(wrapper.find(Image).at(0)).toHaveProp('src', imageIcon);
  });

  it('has image component with addIcon', () => {
    expect(wrapper.find(Image).at(1)).toHaveProp('src', addIcon);
  });

  it('has image component with deleteIcon', () => {
    expect(wrapper.find(Image).at(2)).toHaveProp('src', deleteIcon);
  });

  it('has copy "Add images"', () => {
    expect(wrapper.find('span').at(0)).toHaveText('Add images');
  });

  it('has copy "Add row/column"', () => {
    expect(wrapper.find('span').at(1)).toHaveText('Add row/column');
  });

  it('has copy "Delete row/column"', () => {
    expect(wrapper.find('span').at(2)).toHaveText('Delete row/column');
  });
});
