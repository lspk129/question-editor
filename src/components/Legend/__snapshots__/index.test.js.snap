// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend component renders correctly 1`] = `
<Legend>
  <styled.div>
    <div
      className="sc-bxivhb gDVEBz"
    >
      <h3>
        Legend
      </h3>
      <styled.p>
        <p
          className="sc-ifAKCX NRBvW"
        >
          <styled.img
            small={true}
            src="ic_image_black_24px.svg"
          >
            <img
              className="sc-htpNat dMAxOZ"
              src="ic_image_black_24px.svg"
            />
          </styled.img>
           - 
          <styled.span>
            <span
              className="sc-EHOje jArNoe"
            >
              Add images
            </span>
          </styled.span>
        </p>
      </styled.p>
      <styled.p>
        <p
          className="sc-ifAKCX NRBvW"
        >
          <styled.img
            small={true}
            src="ic_add_circle_black_24px.svg"
          >
            <img
              className="sc-htpNat dMAxOZ"
              src="ic_add_circle_black_24px.svg"
            />
          </styled.img>
           - 
          <styled.span>
            <span
              className="sc-EHOje jArNoe"
            >
              Add row/column
            </span>
          </styled.span>
        </p>
      </styled.p>
      <styled.p>
        <p
          className="sc-ifAKCX NRBvW"
        >
          <styled.img
            small={true}
            src="ic_delete_forever_black_24px.svg"
          >
            <img
              className="sc-htpNat dMAxOZ"
              src="ic_delete_forever_black_24px.svg"
            />
          </styled.img>
           - 
          <styled.span>
            <span
              className="sc-EHOje jArNoe"
            >
              Delete row/column
            </span>
          </styled.span>
        </p>
      </styled.p>
      <styled.p
        note={true}
      >
        <p
          className="sc-ifAKCX ibuCQc"
        >
          <i>
            Italic
          </i>
           
          <styled.span>
            <span
              className="sc-EHOje jArNoe"
            >
              text is editable
            </span>
          </styled.span>
        </p>
      </styled.p>
    </div>
  </styled.div>
</Legend>
`;
